keps/407-gang-scheduling/README.md

```diff
@@ -0,0 +1,407 @@
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP aims to enable gang scheduling support for LeaderWorkerSet by creating a PodGroup for each replica.
+This enhancement allows custom schedulers(e.g. Volcano, coscheduling scheduler-plugin, YuniKorn etc.) to perform 
+gang scheduling on the leader and its worker pods as a single unit, ensuring they are scheduled together to
+avoid deadlocks.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Without gang scheduling support, partial scheduling may lead to resource deadlocks. 
+As reported in [issue #167](https://github.com/kubernetes-sigs/lws/issues/167), when cluster resources are limited, 
+the scheduler may prioritize scheduling leader pods while leaving their worker pods pending. 
+This results in a situation where no inference service can serve properly, as each service requires both leader and worker pods to be running.
+Gang scheduling support would ensure that pods within a replica are either all scheduled together or not scheduled at all, 
+thus preventing resource deadlocks and maintaining service availability.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Create different PodGroup resources for each lws replica for different custom schedulers (Volcano, scheduler-plugins, YuniKorn etc.) to enable gang scheduling
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Implement specific gang scheduling policies
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+LeaderWorkerSet will create a PodGroup resource for each replica and handle the association between pods and PodGroup. When creating or updating a LeaderWorkerSet, the controller will:
+
+1. Create PodGroup resources for each replica
```

Comment:
What does "each replica" refer to in here?. All the pods including leader and workers?

Reply:
Yes, 1 leader pod + (size-1) worker pods is a group, and it means each replica